t/APISIX.pm

```diff
@@ -506,8 +503,10 @@ _EOC_
     end
 
     require "resty.core"
-
-    $extra_init_by_lua_start
+_EOC_
+    if (index($extra_init_by_lua, "apisix.discovery.tars.schema") != -1) {
```

Comment:
Switching the extra_init_by_lua position according to its content is not a good idea. If we hit the problem with some other module like `apisix.blah.blah`, it is too hack to extend the checklist.

We should dig deeper and solve the real problem.

Reply:
Ok,let me think sometime.

Reply:
I find tars test $extra_init_by_lua need to execute before apisix init, so i add a '$extra_init_before_apisix_init' token identify it. WDYT?